### PR TITLE
package changes

### PR DESCRIPTION
**Pull request by bgruening**
Add new `request` python package that is needed here: https://github.com/bgruening/galaxytools/commit/ab72354ddd4ba35612d811c6185769121f50c735

### CODE REVIEW

File: packages/package_python_2_7_deeptools_1_5_11/tool_dependencies.xml [hunk 1]
```diff
@@ -30,13 +30,10 @@
                 <action type="setup_python_environment">
                    <repository name="package_python_2_7_10" owner="iuc">
                        <package name="python" version="2.7.10" />
```
Comment (nsoranzo) [inline packages/package_python_2_7_deeptools_1_5_11/tool_dependencies.xml]: The previous 2 lines seems to be incorrectly indented.

File: packages/package_request_2_7/tool_dependencies.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
+<tool_dependency>
+    <package name="request" version="2.7">
```
Comment (nsoranzo) [inline packages/package_request_2_7/tool_dependencies.xml]: s/request/requests

File: packages/package_request_2_7/.shed.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+categories:
+- Tool Dependency Packages
+description: Contains a tool dependency definition that downloads and compiles version
+  2.7 of the python request library
```
Comment (nsoranzo) [inline packages/package_request_2_7/.shed.yml]: s/request/requests/

File: packages/package_request_2_7/.shed.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+long_description: |
+  Requests is an Apache2 Licensed HTTP library, written in Python, for human beings.
+  Most existing Python modules for sending HTTP requests are extremely verbose and cumbersome.
+  Python’s builtin urllib2 module provides most of the HTTP capabilities you should need, but the api is thoroughly broken.
+  It requires an enormous amount of work (even method overrides) to perform the simplest of tasks.
+name: package_request_2_7
```
Comment (nsoranzo) [inline packages/package_request_2_7/.shed.yml]: s/request/requests/

File: packages/package_request_2_7/.shed.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+owner: iuc
+remote_repository_url: https://github.com/galaxyproject/tools-iuc/tree/master/packages/package_request_2_7
```
Comment (nsoranzo) [inline packages/package_request_2_7/.shed.yml]: s/request/requests/

File: packages/package_request_2_7/tool_dependencies.xml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+                </action>
+                <action type="set_environment">
+                    <environment_variable action="prepend_to" name="PYTHONPATH">$INSTALL_DIR/lib/python</environment_variable>
+                    <environment_variable action="prepend_to" name="PATH">$INSTALL_DIR/bin</environment_variable>
+                    <environment_variable action="set_to" name="PYTHONPATH_REQUEST">$INSTALL_DIR/lib/python</environment_variable>
+                    <environment_variable action="set_to" name="REQUEST_ROOT_DIR">$INSTALL_DIR</environment_variable>
+                </action>
+            </actions>
+        </install>
+        <readme>Compiling and installing the Python request library.</readme>
```
Comment (nsoranzo) [inline packages/package_request_2_7/tool_dependencies.xml]: s/request/requests/

File: packages/package_request_2_7/tool_dependencies.xml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+                <action type="shell_command">
+                    export PYTHONPATH=$PYTHONPATH:$INSTALL_DIR/lib/python &amp;&amp; 
+                    export ATLAS=$ATLAS_ROOT_PATH &amp;&amp;
+                    python setup.py install --install-lib $INSTALL_DIR/lib/python --install-scripts $INSTALL_DIR/bin
+                </action>
+                <action type="set_environment">
+                    <environment_variable action="prepend_to" name="PYTHONPATH">$INSTALL_DIR/lib/python</environment_variable>
+                    <environment_variable action="prepend_to" name="PATH">$INSTALL_DIR/bin</environment_variable>
+                    <environment_variable action="set_to" name="PYTHONPATH_REQUEST">$INSTALL_DIR/lib/python</environment_variable>
+                    <environment_variable action="set_to" name="REQUEST_ROOT_DIR">$INSTALL_DIR</environment_variable>
```
Comment (nsoranzo) [inline packages/package_request_2_7/tool_dependencies.xml]: s/REQUEST/REQUESTS/ on the previous 2 lines.

**Comment by bgruening** [comment]
@nsoranzo fixed!

File: packages/package_python_2_7_deeptools_1_5_11/tool_dependencies.xml (hunk 1 above)
Comment (nsoranzo) [inline packages/package_python_2_7_deeptools_1_5_11/tool_dependencies.xml]: @bgruening Sorry, these are still incorrectly indented.

File: packages/package_requests_2_7/.shed.yml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+categories:
+- Tool Dependency Packages
+description: Contains a tool dependency definition that downloads and compiles version
+  2.7 of the python requests library
+long_description: |
+  Requests is an Apache2 Licensed HTTP library, written in Python, for human beings.
+  Most existing Python modules for sending HTTP requests are extremely verbose and cumbersome.
+  Python’s builtin urllib2 module provides most of the HTTP capabilities you should need, but the api is thoroughly broken.
```
Comment (nsoranzo) [inline packages/package_requests_2_7/.shed.yml]: s/api/API/

**Comment by nsoranzo** [comment]
Should `package_requests_2_7` depend on `package_python_2_7`?

Also, should it be called `package_python_2_7_requests_2_7`?

**Comment by bgruening** [comment]
@nsoranzo Python2.7 is currently not required for the requests library. The new `package_python_2_7_*` packages which strictly depends on a specific Python version to enable the use of Python3 libraries are not working currently without setting `enable_beta_tool_command_isolation = True`. This has the downside that all internal converter tools that are importing eggs to resolve dependencies are not working. :(

**Comment by nsoranzo** [comment]
@bgruening I'm concerned that if we don't require `package_python_2_7` we will have this problem on Ubuntu 14.04 and probably other distros: http://stackoverflow.com/questions/29134512/insecureplatformwarning-a-true-sslcontext-object-is-not-available-this-prevent

To fix this we need to install `requests[security]` instead of plain `requests`, i.e. depend on pyOpenSSL, ndg-httpsclient and pyasn1.

**Comment by bgruening** [comment]
I would like to avoid the openssl dependency hell. Does a old version of request work for you as well? requests==2.5.3 should not have this problem ...

**Comment by nsoranzo** [comment]
requests <= 2.5.3 has security issue CVE-2015-2296

**Comment by bgruening** [comment]
I know but we only use it for retrieving data. Not sure what the best way is. My preferred way is to bind it to python2.7, but this method is not ready yet until we get the metadata calculation separated from the Galaxy process.

**Comment by nsoranzo** [comment]
But other packages can depend on it later on and will be vulnerable. I still think it is better to depend on `package_python_2_7` for this one.

**Comment by bgruening** [comment]
Ok, the package depends now on Python 2.7. Sorry had forgotten to commit the indentation fix.

File: packages/package_python_2_7_requests_2_7/tool_dependencies.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    <package name="python" version="2.7">
+        <repository name="package_python_2_7" owner="iuc" prior_installation_required="True" />
+    </package>
+    <package name="requests" version="2.7">
+        <install version="1.0">
+            <actions>
+                <action type="setup_python_environment">
+                   <repository name="package_python_2_7" owner="iuc">
+                       <package name="python" version="2.7" />
+                   </repository>
```
Comment (nsoranzo) [inline packages/package_python_2_7_requests_2_7/tool_dependencies.xml]: Wrong indentation for the previous 3 lines.

**Comment by nsoranzo** [comment]
Thanks @bgruening!

**Comment by bgruening** [comment]
@nsoranzo updated it again, this time also with a fix for: https://github.com/galaxyproject/tools-iuc/issues/289

**Comment by nsoranzo** [comment]
Thanks @bgruening!

**Pull request merged**